Mathlib/LinearAlgebra/SesquilinearForm.lean

```diff
@@ -135,36 +137,38 @@ theorem ortho_smul_right {B : V₁ →ₛₗ[I₁] V₂ →ₛₗ[I₂] K} {x y}
     · exact H
 #align linear_map.ortho_smul_right LinearMap.ortho_smul_right
 
-/-- A set of orthogonal vectors `v` with respect to some sesquilinear form `B` is linearly
+/-- A set of orthogonal vectors `v` with respect to some sesquilinear map `B` is linearly
   independent if for all `i`, `B (v i) (v i) ≠ 0`. -/
-theorem linearIndependent_of_isOrthoᵢ {B : V₁ →ₛₗ[I₁] V₁ →ₛₗ[I₁'] K} {v : n → V₁}
+theorem linearIndependent_of_isOrthoᵢ {B : V₁ →ₛₗ[I₁] V₁ →ₛₗ[I₁'] V} {v : n → V₁}
     (hv₁ : B.IsOrthoᵢ v) (hv₂ : ∀ i, ¬B.IsOrtho (v i) (v i)) : LinearIndependent K₁ v := by
   classical
     rw [linearIndependent_iff']
     intro s w hs i hi
     have : B (s.sum fun i : n ↦ w i • v i) (v i) = 0 := by rw [hs, map_zero, zero_apply]
-    have hsum : (s.sum fun j : n ↦ I₁ (w j) * B (v j) (v i)) = I₁ (w i) * B (v i) (v i) := by
+    have hsum : (s.sum fun j : n ↦ I₁ (w j) • B (v j) (v i)) = I₁ (w i) • B (v i) (v i) := by
       apply Finset.sum_eq_single_of_mem i hi
       intro j _hj hij
-      rw [isOrthoᵢ_def.1 hv₁ _ _ hij, mul_zero]
-    simp_rw [B.map_sum₂, map_smulₛₗ₂, smul_eq_mul, hsum] at this
+      rw [isOrthoᵢ_def.1 hv₁ _ _ hij, smul_zero]
+    simp_rw [B.map_sum₂, map_smulₛₗ₂, hsum] at this
     apply (map_eq_zero I₁).mp
-    exact eq_zero_of_ne_zero_of_mul_right_eq_zero (hv₂ i) this
+    exact Or.elim (smul_eq_zero.mp this)
+      (fun y ↦ by simpa using y)
+      (fun hfalse ↦ False.elim $ (hv₂ i) hfalse)
```

Comment:
The first one is just `id`, but `LinearMap.id` is not protected. The second is just fun new syntax in Lean 4.
```suggestion
    exact (smul_eq_zero.mp this).elim _root_.id (hv₂ i · |>.elim)
```